Library/Homebrew/diagnostic.rb

```diff
@@ -558,15 +569,34 @@ def check_brew_git_origin
         examine_git_origin(HOMEBREW_REPOSITORY, Homebrew::EnvConfig.brew_git_remote)
       end
 
-      def check_coretap_git_origin
-        examine_git_origin(CoreTap.instance.path, Homebrew::EnvConfig.core_git_remote)
+      def check_coretap_integrity
+        coretap = CoreTap.instance
+        if broken_tap?(coretap)
+          <<~EOS
+            Homebrew/homebrew-core was not tapped properly.
+            To fix:
+              rm -rf "#{coretap.path}"
+              brew tap homebrew/core
+          EOS
+        else
```

Comment:
```suggestion
          EOS
          return
        end
```

which saves the need for the `else`/indent (and same below)